float32 value_min
float32 value_max
